react/Header/NotificationsLink/NotificationsLinkContainer.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getCookieFromString } from './utils';
+import { AUTHENTICATED } from '../../private/authStatusTypes';
+import { EXPERIMENT_ID, AB_CRUNCH_URL } from './constants';
+import NotificationsLink from './NotificationsLink';
+
+class NotificationsLinkContainer extends React.Component {
+  static propTypes = {
+    authenticationStatus: PropTypes.bool,
+    linkRenderer: PropTypes.element
+  };
+
+  state = {
+    isInExperiment: false
+  };
+
+  componentDidMount() {
+    let visitorId = null;
+
+    try {
+      if (typeof document !== 'undefined') {
+        visitorId = getCookieFromString('JobseekerVisitorId', document.cookie);
+      }
+      if (visitorId) {
+        const url = `${AB_CRUNCH_URL}/${visitorId}`;
+
+        fetch(url)
+          .then(response => response.json())
+          .then(response => {
+            const isInExperiment = Boolean(response.experiments[EXPERIMENT_ID]);
+
+            if (isInExperiment) {
+              this.setState({ isInExperiment: true });
+            }
+          })
+          .catch(() => {});
```

Comment:
There are two catches here that swallow errors. Why are they necessary? And is there any way to document this in code... perhaps just an inline comment?

Reply:
The swallowing is just to hide the errors from product teams as there's nothing to worry about from their perspective. The reason for two is because synchronous try catch don't catch promise based exceptions (without async/await).